src/server/active_tran_server.cpp

```diff
@@ -72,7 +72,11 @@ active_tran_server::on_boot ()
 	  std::bind (&active_tran_server::push_request, std::ref (*this), tran_to_page_request::SEND_LOG_PRIOR_LIST,
 		     std::placeholders::_1);
 
-  log_Gl.m_prior_sender.add_sink (m_prior_sender_sink_hook_func);
+  if (log_Gl.m_prior_sender == nullptr)
```

Comment:
Add functions to initialize/finalize must be added to `log_global` (similar to `log_global::..._log_prior_receiver ()`) and these need to be called properly:
- log prior sender must be initialized for ATS (to dispatch transactional log prior lists towards PS - `active_tran_server::on_boot`)
- and PS (to forward transactional log lists towards PTS's - see usage in `page_server::connection_handler::prior_sender_sink_hook ` and `log_pack_log_boot_info`)
Do the initialization/finalization in `server_type.cpp`.
Also, fill in with finalize of prior sender wherever needed similar to prior receiver (see `xboot_shutdown_server`).